test/integration/EndToEnd.t.sol

```diff
@@ -234,6 +236,10 @@ contract TestEndToEnd is Test {
                 kind: uint8(VaultUpdateKind.DeployAndLink)
             }).serialize()
         );
+
+        vm.stopPrank();
+        vm.deal(address(this), DEFAULT_SUBSIDY);
+        cv.gateway.subsidizePool{value: DEFAULT_SUBSIDY}(poolId);
```

Comment:
Maybe it's interesting to put this per test case, to add the exact required gas, in order to ensure the messages are correctly paid as expected. Otherwise, we could not notice some breaking change in the payment